Framework/Muon/src/ApplyMuonDetectorGrouping.cpp

```diff
@@ -0,0 +1,503 @@
+#include "MantidMuon/ApplyMuonDetectorGrouping.h"
+#include "MantidMuon/MuonAlgorithmHelper.h"
+
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Workspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+#include "MantidDataObjects/TableWorkspace.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/ListValidator.h"
+
+#include <algorithm>
+#include <ctype.h>
```

Comment:
In C++ we should avoid using the `.h` variants of headers. Is this for `isalnum`? If so I think the correct header is `<cctype>`